src/decode/lzma2.rs

```diff
@@ -138,24 +147,24 @@ where
 
             lzma_info!("Properties {{ lc: {}, lp: {}, pb: {} }}", lc, lp, pb);
         } else {
-            lc = decoder.lc;
-            lp = decoder.lp;
-            pb = decoder.pb;
+            lc = decoder.lzma_props.lc;
+            lp = decoder.lzma_props.lp;
+            pb = decoder.lzma_props.pb;
```

Comment:
Nit: This block could directly return an `LzmaProperties` struct, rather than extracting the properties here to re-merge them just afterwards.

Reply:
https://github.com/gendx/lzma-rs/pull/73/commits/5716a6bca5872c33ff806ccfd6a87b6ea90e31b9#diff-aec280556b32db2cd7e0d9b67090c0833582fa5e5dac272202f84adcfd64e887R119

Moved the in-flight lc/lp/pb values inside of the `reset_props` blocks as well.